mongo/errors.go

```diff
@@ -561,15 +597,27 @@ func (bwe BulkWriteException) Error() string {
 
 // HasErrorCode returns true if any of the errors have the specified code.
 func (bwe BulkWriteException) HasErrorCode(code int) bool {
-	if bwe.WriteConcernError != nil && bwe.WriteConcernError.Code == code {
-		return true
+	return hasErrorCode(bwe, code)
+}
+
+// ErrorCodes returns a list of error codes returned by the server.
+func (bwe BulkWriteException) ErrorCodes() []int {
+	errorCodeSet := make(map[int]struct{})
+	for _, writeError := range bwe.WriteErrors {
+		errorCodeSet[writeError.Code] = struct{}{}
 	}
-	for _, we := range bwe.WriteErrors {
-		if we.Code == code {
-			return true
-		}
+
+	if bwe.WriteConcernError != nil {
+		errorCodeSet[bwe.WriteConcernError.Code] = struct{}{}
 	}
-	return false
+
+	// Deduplicate error codes.
```

Comment:
Not that it affects us :), but: I see the same logic here and in WriteException’s method. This seems like the sort of thing you’d deduplicate.